src/controllers/dynakube/activegate/internal/statefulset/reconciler.go

```diff
@@ -171,6 +171,23 @@ func (r *Reconciler) deleteStatefulSetIfOldLabelsAreUsed(desiredSts *appsv1.Stat
 	return false, nil
 }
 
+func isUsingOldLabels(currentStatefulSet *appsv1.StatefulSet) bool {
```

Comment:
shouldn't we check the version label if it changed?
we store the version of the activegate in the labels, and if there is a new version, label changes, get redeployed
was this even expected behaviour ?

Reply:
if the version changed, that would affect the calculated hash value so we already detect that change.
this check specifically targets the old values of the statefulset since the labels are also used as match labels which are immutable. In that case we can't update the sts but have to delete it.

however, this also means that, if the version changes the label changes as well.
this has yet to be reflected by this PR, meaning the old as well as the new labels have to be checked, however for the new ones the value has to be checked as well

Reply:
changed it to simply checking the selector instead of the labels